pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentStickyKeyDispatcherMultipleConsumers.java

```diff
@@ -260,6 +267,12 @@ protected void sendMessagesToConsumers(ReadType readType, List<Entry> entries) {
             // readMoreEntries should run regardless whether or not stuck is caused by
             // stuckConsumers for avoid stopping dispatch.
             readMoreEntries();
+        }  else if (currentThreadKeyNumber == 0) {
+            topic.getBrokerService().executor().schedule(() -> {
+                synchronized (PersistentStickyKeyDispatcherMultipleConsumers.this) {
+                    readMoreEntries();
+                }
+            }, 100, TimeUnit.MILLISECONDS);
```

Comment:
One detail is that before this PR there was a bug that if any connection was backpressured, readMoreEntries wouldn't be called and that would block all progress. 

The consequence of this change seems to be that readMoreEntries is called twice when a Key_Shared subscription has at least 2 consumers. That doesn't seem to make sense to me. It seems that #16812 was added to prevent duplicate calls happening at the same time so I guess that is mitigated.